lib/ae_mdw/node/db.ex

```diff
@@ -14,12 +14,61 @@ defmodule AeMdw.Node.Db do
   @type pubkey() :: <<_::256>>
   @type hash_type() :: nil | :key | :micro
   @type height_hash() :: {hash_type(), pos_integer(), binary()}
+  @opaque key_block() :: tuple()
+  @opaque micro_block() :: tuple()
 
   @spec get_blocks(Blocks.block_hash(), Blocks.block_hash()) :: tuple()
   def get_blocks(kb_hash, next_kb_hash) do
     {:aec_db.get_block(kb_hash), get_micro_blocks(next_kb_hash)}
   end
 
+  @spec get_blocks_per_height(Blocks.height(), Blocks.block_hash() | Blocks.height()) :: [
+          {Blocks.height(), [micro_block()], Blocks.block_hash()}
+        ]
+  def get_blocks_per_height(from_height, block_hash) when is_binary(block_hash),
+    do: get_blocks_per_height(from_height, block_hash, nil)
+
+  def get_blocks_per_height(from_height, to_height) when is_integer(to_height) do
+    {:ok, header} = :aec_chain.get_key_header_by_height(to_height + 1)
+
+    last_mb_hash = :aec_headers.prev_hash(header)
```

Comment:
There are some "empty" generations (without microblocks).

Reply:
this is contemplated by the `get_blocks_per_height`, mbs will be an empty array